packages/toolpad-app/src/toolpad/AppEditor/PageEditor/RenderPanel/NodeHud.tsx

```diff
@@ -180,8 +183,20 @@ export default function NodeHud({
           >
             {component?.displayName || '<unknown>'}
             <DragIndicatorIcon color="inherit" />
+            <IconButton
+              aria-label="Duplicate element"
+              color="inherit"
+              onClick={onDuplicate}
+              onMouseDown={stopPropagationHandler}
```

Comment:
Probably no need for this `onMouseDown`? Unless you found some bug with this.
Also I see the delete button is using `onMouseUp` instead of `onClick` - that might be to avoid certain conflicts with the drag & drop / resizing, so not sure if could just use `onMouseUp`?